src/main/java/io/vertx/core/impl/WorkerExecutorImpl.java

```diff
@@ -56,7 +56,7 @@ public WorkerPool getPool() {
   }
 
   @Override
-  public <T> Future<@Nullable T> executeBlocking(Handler<Promise<T>> blockingCodeHandler, boolean ordered) {
+  public synchronized <T> Future<@Nullable T> executeBlocking(Handler<Promise<T>> blockingCodeHandler, boolean ordered) {
     if (closed) {
```

Comment:
the sync should only wrap the if (closed) statement

Reply:
ping @dano

Reply:
Sorry, @vietj  I am out of town this week, and won't be able to update the PR until this weekend.